diff --git a/src/meson.build b/src/meson.build
index b9fca8d..059393a 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,13 +1,13 @@
 gworldscene_api_version = '0.1'
 gobject_introspection_dep = dependency('gobject-introspection-1.0')
-gobject_introspection_girdir = gobject_introspection_dep.get_variable(pkgconfig: 'girdir')
-gobject_introspection_typelibdir = gobject_introspection_dep.get_variable(pkgconfig: 'typelibdir')
-gobject_introspection_libdir = gobject_introspection_dep.get_variable(pkgconfig: 'libdir')
-gobject_introspection_includedir = gobject_introspection_dep.get_variable(pkgconfig: 'includedir')
-gobject_introspection_datadir = gobject_introspection_dep.get_variable(pkgconfig: 'datadir')
-gworldscene_headerdir = gobject_introspection_includedir / ('gworldscene-' + gworldscene_api_version) / 'gworldscene'
-gworldscene_pkgconfigdir = gobject_introspection_libdir / 'pkgconfig'
-gworldscene_vapidir = gobject_introspection_datadir / 'vala' / 'vapi'
+gworldscene_includedir = get_option('includedir')
+gworldscene_libdir = get_option('libdir')
+gworldscene_datadir = get_option('datadir')
+gobject_introspection_girdir = gworldscene_datadir / 'gir-1.0'
+gobject_introspection_typelibdir = gworldscene_libdir / 'girepository-1.0'
+gworldscene_headerdir = gworldscene_includedir / ('gworldscene-' + gworldscene_api_version) / 'gworldscene'
+gworldscene_pkgconfigdir = gworldscene_libdir / 'pkgconfig'
+gworldscene_vapidir = gworldscene_datadir / 'vala' / 'vapi'
 
 gworldscene_deps = [
   dependency('gtk4', version: '>=4.10'),
@@ -54,7 +54,7 @@ gworldscene_lib = shared_library(
   gworldscene_sources,
   dependencies: gworldscene_deps,
   include_directories: gworldscene_inc,
-  install_dir: gobject_introspection_libdir,
+  install_dir: gworldscene_libdir,
   install: true,
   version: meson.project_version(),
   soversion: '0',
@@ -111,7 +111,7 @@ pkg.generate(
   filebase: 'gworldscene-' + gworldscene_api_version,
   description: 'GTK4 OpenGL world scene widget',
   subdirs: [],
-  extra_cflags: ['-I' + (gobject_introspection_includedir / ('gworldscene-' + gworldscene_api_version))],
+  extra_cflags: ['-I' + (gworldscene_includedir / ('gworldscene-' + gworldscene_api_version))],
   install_dir: gworldscene_pkgconfigdir,
   requires: ['gtk4', 'gdk-pixbuf-2.0', 'gio-2.0', 'epoxy', 'gdal', 'assimp', 'libsoup-3.0', 'zlib'],
   version: meson.project_version(),
